bindings/python/cstest_py/src/cstest_py/cstest.py

```diff
@@ -229,42 +229,34 @@ def __init__(self, expected_dict: dict):
 
     def compare(self, actual_insns: list[CsInsn], bits: int) -> TestResult:
         if len(actual_insns) != len(self.insns):
-            log.error(
+            raise Exception(
                 "Number of decoded instructions don't match (actual != expected): "
                 f"{len(actual_insns)} != {len(self.insns):#x}"
             )
-            return TestResult.FAILED
+
         for a_insn, e_insn in zip(actual_insns, self.insns):
-            if not compare_asm_text(
-                a_insn,
-                e_insn.get("asm_text"),
-                bits,
-            ):
-                return TestResult.FAILED
+            def _check(res: bool):
+                if not res:
+                    raise Exception(f"Failed instruction: {a_insn}")
```

Comment:
Why raising an exception instead of returning `TestResult.FAILED`?
This would stop other tests from execute.

Reply:
If you want to cancel the test run at the first failing case, please make it optional via some flag (see `cstest.py::parse_args()`).